pkg/types/openstack/validation/platform.go

```diff
@@ -61,7 +62,26 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
 
 	for _, ip := range p.ExternalDNS {
 		if err := validate.IP(ip); err != nil {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("ExternalDNS"), p.ExternalDNS, err.Error()))
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("externalDNS"), p.ExternalDNS, err.Error()))
+		}
+	}
+
+	if p.APIVIP != "" {
+		if err := validate.IP(p.APIVIP); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, err.Error()))
+		}
+		if !n.MachineNetwork[0].CIDR.Contains(net.ParseIP(p.APIVIP)) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("apiVIP"), p.APIVIP, "IP is not in the machineNetwork"))
```

Comment:
the 2 step check is same for API and ingress VIP, so maybe use one function for both..